scipy/interpolate/src/__fitpack.cc

```diff
@@ -418,9 +420,152 @@ fpknot(const double *x_ptr, ssize_t m,
     }
 
     // round up, like Dierckx does? This is really arbitrary though.
-    ssize_t idx_newknot = (ix[idx_max] + ix[idx_max+1] + 1) / 2;
+    d_ssize_t idx_newknot = (ix[idx_max] + ix[idx_max+1] + 1) / 2;
 
     return x(idx_newknot);
 }
 
+
+
+/*
+ * Evaluate the spline function
+*/
+void
+_evaluate_spline(
+    const double *tptr, d_ssize_t len_t,         // t, shape (len_t,)
+    const double *cptr, d_ssize_t n, d_ssize_t m,  // c, shape (n, m)
+    d_ssize_t k,
+    const double *xp_ptr, d_ssize_t s,           // xp, shape (s,)
+    d_ssize_t nu,
+    int extrapolate,
+    double *out_ptr,                           // out, shape (s, m) NOT CHECKED
+    double *wrk                                // scratch, shape (2k+2,)
+)
+{
+    auto t = ConstRealArray1D(tptr, len_t);
+    auto c = ConstRealArray2D(cptr, n, m);
+    auto xp = ConstRealArray1D(xp_ptr, s);
+    auto out = RealArray2D(out_ptr, s, m);
+
+    d_ssize_t interval = k;
+    for(d_ssize_t ip=0; ip < s; ip++) {
+
+        double xval = xp(ip);
+        interval = _find_interval(t.data, len_t, k, xval, interval, extrapolate);
+        if (interval < 0) {
+            // xval was nan etc
+            for (d_ssize_t jp=0; jp < m; jp++) {
+                out(ip, jp) = std::numeric_limits<double>::quiet_NaN();
+            }
+        }
+        else {
+            // Evaluate (k+1) b-splines which are non-zero on the interval.
+            // on return, first k+1 elements of work are B_{m-k},..., B_{m}
+            _deBoor_D(t.data, xval, k, interval, nu, wrk);
+
+            // Form linear combinations
+            for (d_ssize_t jp=0; jp < m; jp++) {
+                out(ip, jp) = 0.0;
```

Comment:
Maybe 0.0 initialize out outside the loop so this is more prefetchable?

Reply:
Hm. So you're saying that a separate loop zeroing out the buffer is preferable because of... what's more prefetchable?

Reply:
Meaning that the loop does not interrupt. And also faster to zero initialize. But I guess these are micro optimizations that are not worth it. Yeah, probably just nitpicks come to think of it.

Reply:
That's actually interesting :-). I'll try to do some time measurements in a bit--- will be happy when this graduates to where there kinds of optimizations are the main concern. Thank you!